src/modules/secrets/Elsa.Secrets/Providers/SecretNamesOptionsProvider.cs

```diff
@@ -27,7 +27,7 @@ public class SecretNamesOptionsProvider : IActivityPropertyOptionsProvider
             var secrets = secretsManager.GetSecrets().Result;
 
             var items = secrets.Where(x => string.IsNullOrWhiteSpace(x.Name) == false)
-                .Select(x => new SelectListItem($"{x.DisplayName ?? x.Name} ({x.Type})", $"{x.Type}:{x.Name!}")).ToList();
```

Comment:
The DisplayName is being set as the type or something similar, and the Name field is what is being populated with the users given name.

Reply:
Then perhaps the right fix would be to set the DisplayName to the value that the user sets? Conventionally, the Name property is intended to be a technical name, and DisplayName should be a user defined friendly name. I know you probably didn’t implement it like that since this module is someone else’s contribution, but maybe you wouldn’t mind changing it?

Reply:
Let me see if I can get time to look at it this afternoon, if it seems simple enough I'll push a new update this evening.

Reply:
Much appreciated!

Reply:
@sfmskywalker I unfortunately had surgery on my shoulder last week, so I will not be able to swap names like discussed for a while.

Reply:
Sorry to hear about your shoulder, hope all went well and that you have a speedy recovery from surgery. No worries about this PR, we can do it whenever you're ready. If you don't mind, I'd like to keep this PR open until then, unless it's a significant issue for you, in which case you'll let me know.

Reply:
Thanks, it's expected to take a few weeks for the recovery, just a tore bicep tendon and rotator cuff. Can we approve this PR, and then create a new one later to fix the name vs displayname issue? That way my team can have a way of knowing which item in the dropdown list is what.

We implemented the oAuth2 functionality of the secrets module in one of our custom activities. It works pretty well, and was impressed with it. We have some improvements we would like to contribute towards its usability/setup in custom activities, we can add the name vs displayname update to that PR when we do it if you like.

Reply:
Yes, let's do it like that 👍🏻